data/kotlin/data-uploader/README.md

```diff
@@ -0,0 +1,34 @@
+# DATA UPLOADER
+This is an application for consuming data exposed by the pub/sub system in the [RADAR-IoT](https://github.com/RADAR-base/RADAR-IoT) framework and processing/uploading to to different destinations.
+The application consumes data from channels in a pub/sub system and can consume it in various ways. One such implementation provided is uploading the data to the [RADAR-base platform's backend](https://github.com/RADAR-base/RADAR-Docker).
+
+## Terminology
+These are the explanation of the main terms used in the context of this application - 
+
+1. `Handler` - This is used for Handling data coming in from the pub/sub system. This is also responsible for subscribing to the topics/channels of the pub/sub system. It usually forwards data to the `DataConsumer`s.
+2. `Consumer` or `DataConsumer` - These are the components that handle the actual processing of the data that is received from the `Handler`s.
+3. `Converter` - These are used to convert the message from one format to another. So that data can be read from the pub/sub system and sent/processed by the destination system(for DataConsumers). Hence these are specific to a particular sensor for a particular consumer. One implementation of a converter can be - deserializing a message received in JSON format from the pub/sub system and serializing it to AVRO format for uploading to kafka.
```

Comment:
used in the consumer?

Reply:
Converters convert the data received from a pub/sub channel to a format suitble for a consumer. So they are the bridge between the DataHandler (which reads from the pub/sub) and the DataConsumers (which do the actual processing of the data,). You can think of it as pre-processing of the data. And Yes converters are injected into the consumers for performing any conversions.